portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -23,10 +26,48 @@
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  private ArrayList<Comments> comments = new ArrayList<>();
+  private ArrayList<String> commentString = new ArrayList<>();
+
+  /**
+   * Converts an ArrayList instance into a JSON string using the Gson library.
+   */
+  private String convertToJsonUsingGson() {
+    Gson gson = new Gson();
+    String json = gson.toJson(commentString);
+    return json;
+  }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello Fiza!</h1>");
+    response.setContentType("application/json;");
+    String json = convertToJsonUsingGson();
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the input from the form.
+    String name = getParameter(request, "fname", "");
+    String comment = getParameter(request, "comment-input", "");
+
+    Comments input = new Comments(name, comment);
+    comments.add(input);
+    commentString.add(comment);
+
+    // Respond with the result.
+    response.sendRedirect("/index.html");
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
```

Comment:
Nice use of your own encapsulation to get different and, for you, a more useful interface. It also has the side benefit of allowing you to get the original behavior by passing **null** as the default value.
I like it.